src/bug-filing/common/issue-filing-controller-impl.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserAdapter } from '../../background/browser-adapter';
+import { BaseStore } from '../../common/base-store';
+import { EnvironmentInfo } from '../../common/environment-info-provider';
+import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
+import { BugServiceProperties, UserConfigurationStoreData } from '../../common/types/store-data/user-configuration-store';
+import { BugFilingServiceProvider } from '../bug-filing-service-provider';
+
+export type IssueFilingController = {
+    fileIssue: (serviceKey: string, issueData: CreateIssueDetailsTextData) => void;
+};
+
+export class IssueFilingControllerImpl implements IssueFilingController {
```

Comment:
Here `IssueFilingControllerImpl` is a class that implements abstract type `IssueFilingController`.  (`StoreActionMessageCreatorImpl` is another example of this naming pattern). We also have `BugFilingServiceProviderImpl` which is an instance of concrete type `BugFilingServiceProvider`. These seem like different concepts that should have different names, rather than both being `Impl`. (I think the usage is good here, not as good for `BugFilingServiceProvider`, but would like to hear what others think...)